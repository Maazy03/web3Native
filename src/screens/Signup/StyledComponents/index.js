import styled from 'styled-components/native';

export const MainScreenLayout = styled.View`
  background: #141414
  display:flex;
  flex-direction:column;
  justify-content:center;
  align-items:center;
  height:100%
  position: relative;
`;

export const RowBreakupLayout = styled.View`
  width: 312px;
  margin-bottom: ${props => (props.mgBottom ? props.mgBottom : '0')}px;
  position: absolute;
  top: ${props => props.top}px;
`;

export const GoogleLoginView = styled.TouchableOpacity`
  width: 312px;
  margin-bottom: ${props => (props.mgBottom ? props.mgBottom : '0')}px;
  position: absolute;
  top: ${props => props.top}px;
  flex:1;
`;

export const BackView = styled(RowBreakupLayout)`
  left: 24px;
`;

export const LoginWithGoogleView = styled(RowBreakupLayout)`
  height: 52px;
`