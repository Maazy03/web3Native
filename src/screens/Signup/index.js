import React from 'react';
import {MainScreenLayout, RowBreakupLayout, BackView, GoogleLoginView} from './StyledComponents';
import {MainHeading, SubHeading} from '../../components/Headings';
import {GoogleButton} from '../../components/Buttons';
import Back from '../../assets/images/back.svg';
import {TouchableOpacity, View} from 'react-native';

export default Signup = ({navigation}) => {

  const handleNavigation = () => {
    navigation.navigate("SignupName");
  };

  return (
    <>
      <MainScreenLayout>
        <BackView top={56}>
          <TouchableOpacity onPress={() => navigation.goBack()}>
            <Back />
          </TouchableOpacity>
        </BackView>

        <RowBreakupLayout top={216}>
          <MainHeading children="Great Choice!" />
        </RowBreakupLayout>

        <RowBreakupLayout top={267}>
          <SubHeading
            children={`Now, letâ€™s register using your google account`}
          />
        </RowBreakupLayout>

        <GoogleLoginView top={400}>
          <GoogleButton children="Register with Google" onPressFunc={handleNavigation} />
        </GoogleLoginView>

      </MainScreenLayout>
    </>
  );
};
