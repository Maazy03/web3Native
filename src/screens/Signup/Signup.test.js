import 'react-native';
import React from 'react';
import Signup from './index';
 
 // Note: test renderer must be required after react-native.
 import renderer from 'react-test-renderer';
 
 it('renders correctly', () => {
   renderer.create(<Signup />);
   const tree = renderer.create( <Signup /> ).toJSON();
  
   expect(tree).toMatchSnapshot();
 });