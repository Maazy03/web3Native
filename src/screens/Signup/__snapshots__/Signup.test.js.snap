// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#141414",
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 0,
          "width": 258,
        },
      ]
    }
    width={258}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2FEAC6",
              "fontFamily": "Rubik-Bold",
              "fontSize": 32,
              "fontStyle": "normal",
              "fontWeight": "500",
              "height": 50,
              "letterSpacing": 1,
              "lineHeight": 42,
              "marginTop": 10,
              "textAlign": "center",
            },
          ]
        }
      >
        Great Choice!
      </Text>
    </View>
  </View>
  <View
    mgBottom={65}
    style={
      Array [
        Object {
          "marginBottom": 65,
          "width": 280,
        },
      ]
    }
    width={280}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fafafa",
              "fontFamily": "Rubik-Regular",
              "fontSize": 20,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "height": 100,
              "letterSpacing": 1,
              "lineHeight": 31,
              "textAlign": "center",
            },
          ]
        }
      >
        Now, letâ€™s register using your google account
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 0,
          "width": 312,
        },
      ]
    }
    width={312}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 24,
              "display": "flex",
              "flexDirection": "row",
              "height": 52,
              "justifyContent": "center",
              "width": 312,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/images/googleLogo.png",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": undefined,
                "width": undefined,
              },
              undefined,
            ]
          }
        />
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/images/googleLogo2.png",
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "color": "#141414",
                "display": "flex",
                "fontFamily": "Rubik-Regular",
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "normal",
                "justifyContent": "center",
                "marginLeft": 15,
              },
            ]
          }
        >
          Register with Google
        </Text>
      </View>
    </View>
  </View>
</View>
`;
